internal/engine.go

```diff
@@ -182,6 +234,13 @@ func (e *Engine) cleanupTemp(temp string) {
 	}
 }
 
+func (e *Engine) applySeverity(issues []tt.Issue, rule LintRule) []tt.Issue {
+	for i := range issues {
+		issues[i].Severity = rule.Severity()
+	}
+	return issues
+}
```

Comment:
Calling the `Severity()` function inside the loop may lead to performance degradation when there are many issues. It would be better to move it outside the loop.

```suggestion
func (e *Engine) applySeverity(issues []tt.Issue, rule LintRule) []tt.Issue {
    severity := rule.Severity()
    for i := range issues {
        issues[i].Severity = severity
    }
    return issues
}
```

Reply:
The other option I see would be to implement it in every rules when creating the issues (went that way before reverting back as I didn't really knew if it was a good idea)

Reply:
https://github.com/gnoverse/tlin/pull/91/commits/178d7974ba0771a82e72b414363dcfe40db64edd